pyfr/plugins/solutionwriter.py

```diff
@@ -0,0 +1,40 @@
+# -*- coding: utf-8 -*-
+
+from collections import OrderedDict
+
+from pyfr.h5writer import H5Writer
+from pyfr.inifile import Inifile
+from pyfr.plugins.base import BasePlugin
+
+
+class SolutionWriterPlugin(BasePlugin, H5Writer):
+    name = 'solutionwriter'
+    systems = ['*']
+
+    def __init__(self, intg, cfgsect, *args, **kwargs):
+        BasePlugin.__init__(self, intg, cfgsect)
```

Comment:
Its not just the right order, its kind of hacky. 
1. Even if `A` and `B` both are similar objects inheriting only from `object`, B calls `super.__init__(a)` whereas `A` does not.
2. There is no way of creating an object of class `B`.

Any way, I have switched to composition. soln-writer will have a _writer memeber object.